### DISCUSSION TITLE
GraalVM compatibility? native image support added in 13.23.1

### DISCUSSION THREAD

**Discussion by jnehlmeier** [ebean-orm/ebean#2845] [General]
Is EBean GraalVM compatible out of the box or does it require additional GraalVM configuration because of reflection? 
If it needs configuration, are you planning to contribute such a configuration regularly to https://github.com/oracle/graalvm-reachability-metadata so it can easily be pulled during builds?

For example using Gradle/Maven you can simply configure it to check for GraalVM configurations for 3rd party libraries: https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html#metadata-support

**Comment by rbygrave** [comment]
> Is EBean GraalVM compatible out of the box or does it require additional GraalVM configuration because of reflection?

Not sure. I haven't got to try it yet (although I have played with GraalVM native-image).

> are you planning to contribute such a configuration regularly to

I'd suggest it would be better to include the configuration in ebean itself. That metadata there is only for projects that don't / won't and I think we could and should.

**Comment by jnehlmeier** [comment]
Sure, having GraalVM configuration files in `META-INF/native-image/` is preferable.

For a smooth out of the box experience 3rd party dependencies of Ebean would also need to have GraalVM configuration files (if they need one). I don't think Ebean should provide them, as you can not know which exact version ends up on classpath. But having a listing of dependencies that need attention could be helpful for developers.

**Reply by rbygrave** [reply]
Support added with https://github.com/ebean-orm/ebean/issues/2895

> listing of dependencies that need attention

The only dependency was avaje-config, and that is now sorted in version 3.9

The improvement that could be made is for the `findDto()` case where reflect-config entry for the dto class is needed as ebean uses reflection to find the constructor and setters and then uses invoke dynamic to populate the dtos.
